### DISCUSSION TITLE
Which Mode do You Prefer?

### DISCUSSION THREAD

**Discussion by caoccao** [caoccao/Javet#92] [General]
As Javet provides 2 modes (Node.js mode and V8 mode), which one do you prefer? And why?

[![](https://api.gh-polls.com/poll/01FG0WB28TQR33ZJ3FBSG9MWNX/Node.js%20mode)](https://api.gh-polls.com/poll/01FG0WB28TQR33ZJ3FBSG9MWNX/Node.js%20mode/vote)
[![](https://api.gh-polls.com/poll/01FG0WB28TQR33ZJ3FBSG9MWNX/V8%20mode)](https://api.gh-polls.com/poll/01FG0WB28TQR33ZJ3FBSG9MWNX/V8%20mode/vote)

**Comment by caoccao** [comment]
Personally, I prefer V8 mode because:

1. It is more secure.
2. It performs better.

Initially, I didn't plan for Node.js mode. Some Javet users asked me if Node.js mode was possible, yes, so I brought Node.js in.

However, V8 mode lacks of some API e.g. setTimeout, setInterval. So I started project [Javenode](https://github.com/caoccao/Javenode) that aims at simulating Node.js with Java in V8 mode.

**Reply by Cormanz** [reply]
Is Javenode a full node polyfill, or just ES6 globals (setTimeout, setInterval, etc etc). We're trying to make V8 use the Deno API with our use-case

**Reply by caoccao** [reply]
Good questions. In the short term, it's for ES5. In the long term, I wish it cover a wide range of Node.js modules. But, the long term goal definitely needs the community.

And, yes, you are welcome contributing Deno API.

**Comment by notsellsunset** [comment]
I prefer Node.js because it can support NPM modules
